### PR TITLE
Modernize code by replacing deprecated headers

### PR DESCRIPTION
**Pull request by SunBlack**
Changes are done by: run-clang-tidy -header-filter='.*' -checks='-*,modernize-deprecated-headers' -fix

### CODE REVIEW

File: apps/src/openni_octree_compression.cpp [hunk 1]
```diff
@@ -47,10 +47,10 @@
 
 #include <iostream>
 #include <vector>
-#include <stdio.h>
+#include <cstdio>
 #include <sstream>
-#include <stdlib.h>
-#include "stdio.h"
+#include <cstdlib>
+#include <cstdio>
```
Comment (taketwo) [inline apps/src/openni_octree_compression.cpp]: Duplicate

File: io/include/pcl/compression/color_coding.h [hunk 2]
```diff
@@ -40,10 +40,10 @@
 #include <iterator>
 #include <iostream>
 #include <vector>
-#include <string.h>
+#include <cstring>
 #include <iostream>
-#include <stdio.h>
-#include <string.h>
+#include <cstdio>
+#include <cstring>
```
Comment (taketwo) [inline io/include/pcl/compression/color_coding.h]: Duplicate

File: io/include/pcl/compression/octree_pointcloud_compression.h [hunk 3]
```diff
@@ -50,10 +50,10 @@
 #include <iterator>
 #include <iostream>
 #include <vector>
-#include <string.h>
+#include <cstring>
 #include <iostream>
-#include <stdio.h>
-#include <string.h>
+#include <cstdio>
+#include <cstring>
```
Comment (taketwo) [inline io/include/pcl/compression/octree_pointcloud_compression.h]: Duplicate

File: io/include/pcl/compression/point_coding.h [hunk 4]
```diff
@@ -40,10 +40,10 @@
 #include <iterator>
 #include <iostream>
 #include <vector>
-#include <string.h>
+#include <cstring>
 #include <iostream>
-#include <stdio.h>
-#include <string.h>
+#include <cstdio>
+#include <cstring>
```
Comment (taketwo) [inline io/include/pcl/compression/point_coding.h]: Duplicate

**Comment by SunBlack** [comment]
I had to remove

https://github.com/PointCloudLibrary/pcl/blob/43b15dd62097c5587b837e35a97aaa9b9e87b82e/common/include/pcl/pcl_macros.h#L173-L179

because it caused compiler issues after changing `math.h` to `cmath` and it seems not to be used anywhere.

**Comment by SunBlack** [comment]
I had to revert changes to https://github.com/PointCloudLibrary/pcl/blob/c8f8df910f29552adbbbb752fe6d2f119f3e0eb8/common/include/pcl/common/fft/kiss_fft.h#L3-L6 because he couldn't found cstdlib. I don't understand why (it's before `extern "C"` and not within).

**Pull request merged**